utils/get_supported_scylla_base_versions.py

```diff
@@ -0,0 +1,78 @@
+
+
+# We support to migrate from specific OOS version to enterprise
+supported_src_oos = {'2021.1': '4.3', '2020.1': '4.0', '2019.1': '3.0'}
+# If new support distro shared repo with others, we need to assign the start support versions. eg: centos8
+start_support_versions = {'centos-8': {'scylla': '4.1', 'enterprise': '2021.1'}}
+
+
+def get_supported_scylla_base_versions(product, version, supported_versions):  # pylint: disable=too-many-branches
+    """
+    We have special base versions list for each release, and we don't support to upgraded from enterprise
+    to opensource. This function is used to get the base versions list which will be used in the upgrade test.
+
+    @product: Scylla product name, scylla or scylla-enterprise
+    @version: The major version of unstable scylla, eg: 4.6.dev, 4.5, 2021.2.dev, 2021.1
+    @supported_version: all scylla release versions, the base versions will be filtered out from the supported_version
+    """
+    oos_base_version = []
+    ent_base_version = []
+
+    oos_release_list = [i for i in supported_versions if not is_enterprise(i)]
+    ent_release_list = [i for i in supported_versions if is_enterprise(i)]
+
+    if product == 'scylla-enterprise':
+        if version in supported_versions:
+            # The dest version is a released enterprise version
+            idx = ent_release_list.index(version)
+            oos_base_version.append(supported_src_oos.get(version))
+            if idx == 0:
+                # The dest version is a minor release of latest enterprise release
+                ent_base_version.append(version)
+            else:
+                # Choose the last release as upgrade base
+                ent_base_version += ent_release_list[idx-1:][:2]
+        elif re.match(r'\d+.\d+', version) and parse_version(version) < parse_version(ent_release_list[0]):
+            # The dest version is smaller than the first supported enterprise release, it might be an invalid dest version
+            pass
+        else:
+            oos_base_version.append(oos_release_list[-1])
+            ent_base_version += ent_release_list[-2:]
+    elif product == 'scylla':
+        if version in supported_versions:
+            # The dest version is a released opensource version
+            idx = oos_release_list.index(version)
+            if idx == 0:
+                # The dest version is a minor release of latest enterprise release
+                oos_base_version.append(version)
+            else:
+                # Choose the last release as upgrade base
+                oos_base_version += oos_release_list[idx-1:][:2]
+        elif re.match(r'\d+.\d+', version) and parse_version(version) < parse_version(oos_release_list[0]):
+            # The dest version is smaller than the first supported opensource release, it might be an invalid dest version
+            pass
+        else:
+            oos_base_version.append(oos_release_list[-1])
```

Comment:
```suggestion
        elif re.match(r'\d+.\d+', version) and parse_version(version) < parse_version(oos_release_list[0]):
            # If dest version is smaller than the first supported opensource release, it might be an invalid dest version
            oos_base_version.append(oos_release_list[-1])
```